src/storm/storage/expressions/PredicateExpression.cpp

```diff
@@ -42,11 +43,47 @@ namespace storm {
 
         std::shared_ptr<BaseExpression const> PredicateExpression::simplify() const {
             std::vector<std::shared_ptr<BaseExpression const>> simplifiedOperands;
+            uint64_t trueCount = 0;
             for (auto const& operand : operands) {
-                simplifiedOperands.push_back(operand->simplify());
+                auto res = operand->simplify();
+                if (res->isLiteral()) {
+                    if (res->isTrue()) {
+                        if (predicate == PredicateType::AtLeastOneOf) {
+                            return res;
+                        } else {
+                            assert(predicate == PredicateType::AtMostOneOf || predicate == PredicateType::ExactlyOneOf);
+                            trueCount++;
+                            simplifiedOperands.push_back(res);
+                        }
+                    } else {
+                        assert (res->isFalse());
+                        assert (predicate == PredicateType::AtMostOneOf || predicate == PredicateType::AtLeastOneOf || predicate == PredicateType::ExactlyOneOf);
+                        // do nothing, in particular, do not add.
+                    }
+                } else {
+                    simplifiedOperands.push_back(res);
+                }
+            }
+            if (trueCount > 1 && (predicate == PredicateType::ExactlyOneOf || predicate == PredicateType::AtMostOneOf)) {
+                return std::shared_ptr<BaseExpression>(new BooleanLiteralExpression(this->getManager(),
+                                                                                    false));
+            }
+
+            if (simplifiedOperands.size() == 0) {
+                switch(predicate) {
+                    case PredicateType::ExactlyOneOf: return std::shared_ptr<BaseExpression>(new BooleanLiteralExpression(this->getManager(),
+                                                                                                                          false));
+                    case PredicateType::AtLeastOneOf: return std::shared_ptr<BaseExpression>(new BooleanLiteralExpression(this->getManager(),
+                                                                                                                          false));
+                    case PredicateType::AtMostOneOf: return std::shared_ptr<BaseExpression>(new BooleanLiteralExpression(this->getManager(),
+                                                                                                                          true));
+                }
             }
```

Comment:
I think the current version does not properly distinguish the two cases `trueCount=0` and `trueCount=1`, right?
This one is more readable, imo:
```suggestion
            if (simplifiedOperands.size() == 0) {
                switch(predicate) {
                    case PredicateType::ExactlyOneOf: return std::shared_ptr<BaseExpression>(new BooleanLiteralExpression(this->getManager(),
                                                                                                                          trueCount == 1));
                    case PredicateType::AtLeastOneOf: return std::shared_ptr<BaseExpression>(new BooleanLiteralExpression(this->getManager(),
                                                                                                                          trueCount >= 1));
                    case PredicateType::AtMostOneOf: return std::shared_ptr<BaseExpression>(new BooleanLiteralExpression(this->getManager(),
                                                                                                                          trueCount <= 1));
                }
            }
```